### PR TITLE
Add new table for user recommendation events

### PR DESCRIPTION
**Pull request by paramsingh**
# Problem

We want users to be able to recommend tracks to their followers.

# Solution

This PR adds a table that would be used to save those recommendation events.

# Action

We can discuss whether we want to do things this way or if we want a more generic table to store feed events.

### CODE REVIEW

File: admin/sql/updates/2021-03-02-add-user-recommendation-event-table.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
+BEGIN;
+
+CREATE TYPE user_recommendation_event_type_enum AS ENUM('recording');
+
+CREATE TABLE recommendation.user_recommendation_event (
+  id                    SERIAL, -- PK
+  user_id               INTEGER, -- FK to "user"
+  recommendation_type   user_recommendation_event_type_enum,
+  metadata              JSONB,
```
Comment (vansika) [inline admin/sql/updates/2021-03-02-add-user-recommendation-event-table.sql:9]: What do you plan to store in here?

**Review by mayhem — COMMENTED** [review]
I think this is pretty good. I like that the data in JSONB and and that there is an enum type defining the event -- that covers all I need. Two things I think we should change/add:

1. I'm not a fan of the name user_recommendation_event since there will be events that won't specifically be about recommendations, but other "events". How about user_timeline_event?
2. This table needs indexes -- we need to carefully consider the index type that we would create on the recommendation_type field since its an enum with few distinct values.

File: admin/sql/updates/2021-03-02-add-user-recommendation-event-table.sql (hunk 1 above)
Comment (paramsingh) [reply admin/sql/updates/2021-03-02-add-user-recommendation-event-table.sql:9]: I haven't defined a spec yet, but for the track recommendation feature it'd store track metadata (artist name, track name, spotify ID, MBIDs) etc.

**Review by mayhem — APPROVED** [review]
As per our IRC convo, the spec for what goes into the metadata field will be provided in the next PR, so I'm ok to merge this.

**Pull request merged**